test/NuGet.Core.Tests/NuGet.Packaging.Test/PackageBuilderTest.cs

```diff
@@ -2485,6 +2487,73 @@ var contentTypesReader
             }
         }
 
+        [Fact]
+        public void Icon_IconMaxFileSizeExceeded_ThrowsException()
+        {
+            TestIconPackaging(
+                new IconTestSourceDirectory("icon.jpg", "icon.jpg", PackageBuilder.MaxIconFileSize + 1),
+                "The icon file size must not exceed 1 megabyte.");
+        }
+
+        [Fact]
+        public void Icon_IconFileEntryNotFound_ThrowsException()
+        {
+            TestIconPackaging(
+                new IconTestSourceDirectory("icon.jpg", "icono.jpg", 100),
+                "No icon <file> element found for <icon> in the nuspec.");
+        }
+
+        [Fact]
+        public void Icon_HappyPath_Suceed()
+        {
+            TestIconPackaging(
+                new IconTestSourceDirectory("icon.jpg", "icon.jpg", 400),
+                string.Empty);
+        }
+
+        [Fact(Skip="Need to solve https://github.com/NuGet/Home/issues/6941 to run this test case")]
+        public void Icon_MultipleIconFilesResolved_ThrowsException()
```

Comment:
I consider this out of scope for the icon work. If someone implements a fix for the issue, the fix should be general and work for icons, licenses, anything else.

Reply:
Let's keep the test. I'm planning to solve the bug.

Reply:
Even if the bug gets fixed, I still don't think this test belongs here. It's unrelated to icons. When duplicate/ambiguous files exist, pack should fail before it even attempts to validate icons, licenses or other content.

Using traditional compiler terminology, I see duplicate files as a syntax error, while checking the licence file is a semantic error. Syntax errors always run before semantic errors.